input/listen.go

```diff
@@ -245,7 +244,10 @@ func (w *udpWorker) consume(l *Listener) {
 		}
 		data := buffer[:b]
 		log.Debugf("%s handler: udp packet from %v (length: %d)", l.kind, src, data)
-		l.handle(data)
+		readErr := l.handleReader(bytes.NewReader(data))
```

Comment:
I would like to avoid heap allocation as much as possible in that hot loop

I would say either add the \n behavior in .handle
or use a sync.Pool

Reply:
@uepoch I don't think there is a need for a Pool here, can you please check if it's fine like this? When I profiled the calls to mallocgc from consume vanished.